sdk/eventhubs/azure-messaging-eventhubs-checkpointstore-blob/src/main/java/com/azure/messaging/eventhubs/checkpointstore/blob/BlobCheckpointStore.java

```diff
@@ -66,22 +68,70 @@ public BlobCheckpointStore(BlobContainerAsyncClient blobContainerAsyncClient) {
         BlobListDetails details = new BlobListDetails().setRetrieveMetadata(true);
         ListBlobsOptions options = new ListBlobsOptions().setPrefix(prefix).setDetails(details);
         return blobContainerAsyncClient.listBlobs(options)
-            // Blob names should be of the pattern eventhub/consumergroup/<partitionId>
+            // Blob names should be of the pattern
+            // fullyqualifiednamespace/eventhub/consumergroup/ownership/<partitionId>
             // While we can further check if the partition id is numeric, it may not necessarily be the case in future.
-            .filter(blobItem -> blobItem.getName().split(BLOB_PATH_SEPARATOR).length == 3)
+            .filter(blobItem -> blobItem.getName().split(BLOB_PATH_SEPARATOR).length == 5)
             .map(this::convertToPartitionOwnership);
     }
 
+    @Override
+    public Flux<Checkpoint> listCheckpoints(String fullyQualifiedNamespace, String eventHubName,
+        String consumerGroup) {
+
+        String prefix = getBlobPrefix(fullyQualifiedNamespace, eventHubName, consumerGroup, CHECKPOINT_PATH);
+        BlobListDetails details = new BlobListDetails().setRetrieveMetadata(true);
+        ListBlobsOptions options = new ListBlobsOptions().setPrefix(prefix).setDetails(details);
+        return blobContainerAsyncClient.listBlobs(options)
+            // Blob names should be of the pattern
+            // fullyqualifiednamespace/eventhub/consumergroup/checkpoints/<partitionId>
+            // While we can further check if the partition id is numeric, it may not necessarily be the case in future.
+            .filter(blobItem -> blobItem.getName().split(BLOB_PATH_SEPARATOR).length == 5)
+            .map(this::convertToCheckpoint);
+    }
+
+    private Checkpoint convertToCheckpoint(BlobItem blobItem) {
+        Checkpoint checkpoint = new Checkpoint();
+
+        logger.info("Found blob for partition {}", blobItem.getName());
+        String[] names = blobItem.getName().split(BLOB_PATH_SEPARATOR);
+        checkpoint.setFullyQualifiedNamespace(names[0]);
+        checkpoint.setEventHubName(names[1]);
+        checkpoint.setConsumerGroup(names[2]);
+        // names[3] is "checkpoint"
+        checkpoint.setPartitionId(names[4]);
+
+        if (CoreUtils.isNullOrEmpty(blobItem.getMetadata())) {
+            logger.warning("No metadata available for blob {}", blobItem.getName());
+            return checkpoint;
+        }
+
+        blobItem.getMetadata().forEach((key, value) -> {
+            switch (key) {
+                case SEQUENCE_NUMBER:
+                    checkpoint.setSequenceNumber(Long.parseLong(value));
+                    break;
+                case OFFSET:
+                    checkpoint.setOffset(Long.parseLong(value));
+                    break;
+                default:
+                    // do nothing, other metadata that we don't use
+                    break;
+            }
```

Comment:
Given that metadata is a `Map<String, String>` would it be better to just attempt to get the value for each of these keys?